src/timeline.js

```diff
@@ -809,6 +844,134 @@ export default class Timeline extends React.Component {
+          (this.state.dragToCreateMode
+            ? startRowObject.getBoundingClientRect().top
+            : currentRowObject.getBoundingClientRect().top) + rowMarginBorder
+        );
+        // get the first selected rows bottom
+        const startBottom = Math.floor(getTrueBottom(startRowObject) - magicalConstant - rowMarginBorder * 2);
+        // the bottom will bleed south unless you counter the margins and boreders from the above rows
+        this._selectBox.start(startX, startBottom);
+        this._selectBox.move(clientX, currentTop);
+      }
+    }
+  }
+
+  #onDragEndSelect() {
+    if (this.state.dragCancel) {
+      // only reset dragCancel on dragend if drag is canceled
+      this.setState({dragCancel: false});
+      return;
+    }
+
+    let {top, left, width, height} = this._selectBox.end();
+    //Get the start and end row of the selection rectangle
+    const topRowObject = getNearestRowObject(left, top);
+    if (topRowObject !== undefined) {
+      // only confirm the end of a drag if the selection box is valid
+      const topRowNumber = Number(getNearestRowNumber(left, top));
+      const topRowLoc = topRowObject.getBoundingClientRect();
+      const rowMarginBorder = getVerticalMarginBorder(topRowObject);
+      const y = Math.floor(topRowLoc.top - rowMarginBorder) + Math.floor(height - rowMarginBorder);
+      const bottomRow = Number(getNearestRowNumber(left + width, adjustRowTopPositionToViewport(topRowObject, y)));
+      //Get the start and end time of the selection rectangle
+      left = left - topRowLoc.left;
+      let startOffset = width > 0 ? left : left + width;
+      let endOffset = width > 0 ? left + width : left;
+      const startTime = getTimeAtPixel(
+        startOffset,
+        this.getStartDate(),
+        this.getEndDate(),
+        this.getTimelineWidth(),
+        this.getTimelineSnap()
+      );
+      const endTime = getTimeAtPixel(
+        endOffset,
+        this.getStartDate(),
+        this.getEndDate(),
+        this.getTimelineWidth(),
+        this.getTimelineSnap()
+      );
+      //Get items in these ranges
+      let selectedItems = [];
+      for (let r = Math.min(topRowNumber, bottomRow); r <= Math.max(topRowNumber, bottomRow); r++) {
+        selectedItems.push(
+          ..._.filter(this.rowItemMap[r], i => {
+            return this.getStartFromItem(i).isBefore(endTime) && this.getEndFromItem(i).isAfter(startTime);
+          })
+        );
+      }
+      !this.state.dragToCreateMode &&
+        this.props.onInteraction &&
+        this.props.onInteraction(Timeline.changeTypes.itemsSelected, selectedItems);
+      if (this.state.dragToCreateMode && this.props.onDragToCreateEnded) {
+        // get avaible itemIndex and call the onDragToCreateEnded
+        const itemIndex = Math.max(...Object.keys(this.itemRowMap)) + 1;
+        this.props.onDragToCreateEnded && this.props.onDragToCreateEnded(topRowNumber, itemIndex, startTime, endTime);
```

Comment:
Nu cred ca mai avea rost sa verifici `this.props.onDragToCreateEnded` pentru ca faci asta in conditia din if. Cred ca puteai sa faci direct `this.props.onDragToCreateEnded(topRowNumber, itemIndex, startTime, endTime);`